diff --git a/python/plugins/processing/algs/help/__init__.py b/python/plugins/processing/algs/help/__init__.py
index 08a14085..166de786 100644
--- a/python/plugins/processing/algs/help/__init__.py
+++ b/python/plugins/processing/algs/help/__init__.py
@@ -43,7 +43,7 @@ def loadShortHelp():
             with codecs.open(filename, encoding='utf-8') as stream:
                 with warnings.catch_warnings():
                     warnings.filterwarnings("ignore", category=DeprecationWarning)
-                    for k, v in yaml.load(stream).items():
+                    for k, v in yaml.safe_load(stream).items():
                         if v is None:
                             continue
                         h[k] = QCoreApplication.translate("{}Algorithm".format(f[:-5].upper()), v)
diff --git a/python/plugins/processing/tests/AlgorithmsTestBase.py b/python/plugins/processing/tests/AlgorithmsTestBase.py
index 2845ac84..ac202983 100644
--- a/python/plugins/processing/tests/AlgorithmsTestBase.py
+++ b/python/plugins/processing/tests/AlgorithmsTestBase.py
@@ -72,7 +72,7 @@ class AlgorithmsTest(object):
         This is the main test function. All others will be executed based on the definitions in testdata/algorithm_tests.yaml
         """
         with open(os.path.join(processingTestDataPath(), self.test_definition_file()), 'r') as stream:
-            algorithm_tests = yaml.load(stream)
+            algorithm_tests = yaml.safe_load(stream)
 
         if 'tests' in algorithm_tests and algorithm_tests['tests'] is not None:
             for idx, algtest in enumerate(algorithm_tests['tests']):
