diff -Naru amarok-master.orig/CMakeLists.txt amarok-master/CMakeLists.txt
--- amarok-master.orig/CMakeLists.txt	2018-05-27 09:44:50.910847230 +0200
+++ amarok-master/CMakeLists.txt	2018-05-27 09:45:44.122844286 +0200
@@ -53,7 +53,7 @@
 set(CMAKE_REQUIRED_INCLUDES "${TAGLIB_INCLUDES}")
 set(CMAKE_REQUIRED_LIBRARIES "${TAGLIB_LIBRARIES}")
 
-check_cxx_source_compiles("#include <asftag.h>
+check_cxx_source_compiles("#include <taglib/asftag.h>
 int main() { TagLib::ASF::Tag tag; return 0;}" TAGLIB_ASF_FOUND)
 if( NOT TAGLIB_ASF_FOUND )
     message(FATAL_ERROR "TagLib does not have ASF support compiled in.")
diff -Naru amarok-master.orig/shared/tag_helpers/ASFTagHelper.h amarok-master/shared/tag_helpers/ASFTagHelper.h
--- amarok-master.orig/shared/tag_helpers/ASFTagHelper.h	2018-05-27 09:44:50.939847228 +0200
+++ amarok-master/shared/tag_helpers/ASFTagHelper.h	2018-05-27 09:45:59.965843410 +0200
@@ -20,7 +20,7 @@
 
 #include "TagHelper.h"
 
-#include <asftag.h>
+#include <taglib/asftag.h>
 
 namespace Meta
 {
