diff -Naru a/boringssl.orig/CMakeLists.txt b/boringssl/CMakeLists.txt
--- a/boringssl.orig/CMakeLists.txt	2018-10-15 20:19:55.669493184 +0200
+++ b/boringssl/CMakeLists.txt	2018-10-15 20:20:32.071491170 +0200
@@ -245,6 +245,8 @@
   set(ARCH "x86")
 elseif (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i386")
   set(ARCH "x86")
+elseif (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i586")
+  set(ARCH "x86")
 elseif (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i686")
   set(ARCH "x86")
 elseif (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "aarch64")
